datalad/customremotes/ria_utils.py

```diff
@@ -117,11 +117,9 @@ def verify_ria_url(url, cfg):
         portdlm=':' if url_ri.port else '',
         port=url_ri.port or '',
     )
-    # this != file is critical behavior, if removed, it will ruin the IO selection
-    # in RIARemote!!
-    return host if protocol != 'file' else None, \
-        url_ri.path if url_ri.path else '/', \
-        url
+    # this ``!= 'file'´´ is critical behavior, if removed, it will ruin the IO
+    # selection in RIARemote!!
+    return host if protocol != 'file' else None, url_ri.path or '/', url
```

Comment:
Reading carefully I see that it is not just reformatting but changing `if` into `or` expression ... and I think it isn't correct since any non empty tuple eg of `None, url_ri.path`.  was this change needed and what was intervention for it?

Reply:
I think that `if url_ri.path` triggers execution iff `url_ri.path` evaluates to `True`. And `url_ri.path or '/'` will evaluate to `url_ri.path` iff `bool(url_ri.path)` evaluates to `True`, else to `'/'`. So I think the two version have identical results. Am I missing something?

The reason for the change was that I was working in the method and thought, it would be a good refactoring, making the code shorter and more readable. The latter is a probably a matter of taste though.

Reply:
I am sorry -- I was misreading the code and was assuming that `,` is taking precedence and making `None, url_ri.path` into `(None, url_ri.path)` tuple.  I should have read original code better -- having them on different lines helped to accent that fact.

With that -- you should be correct with replacing that `if` with `or` here, sorry for the noise.

Reply:
No worries 😉